### ISSUE TITLE
intermittent failed to connect errors when fetching resources via API

### ISSUE THREAD

**Issue by tomasc** [coyote-team/coyote#408]
when accessing the API at `https://staging.coyote.pics/api/v1/resources/[canonical_id]` we are seeing intermittent `failed to connect: Connection reset by peer - SSL_connect` responses

**Comment by sinabahram** [comment]
Tomas, I don’t believe there to be any rate limiting, bt do you have a sense that these go up the faster you request? do they always just work if you request it again, etc.?

 

Trying to figure out what part of the stack is causing this.

**Comment by tomasc** [comment]
@sinabahram not sure – we hit Coyote API via background job – here in parallel, could be 5 at the same time or even more. Best to check your logs.

**Comment by flipsasser** [comment]
This never made it to the Rails app. My guess is you swamped Heroku's meager dynos.

Is this something that will happen regularly, or was this a one-off task?

**Comment by sinabahram** [comment]
Let me know if there’s something I need to do on Heroku to prevent this from happening. Also maybe cloudflare rate limiting this? tomas, can you simply back off just a tad and see if the problem goes away completley?

**Comment by tomasc** [comment]
@flipsasser @sinabahram we ran this on a background job (Sidekiq), which has adjustable parallelism – currently set to 5 on staging. This means you would be hit with up to 5 requests from us at the same time whenever we run complete sync.

We could implement rate limiting, however perhaps it is worth checking your setup first. As even if we limit this job, I can imagine that with multiple apps periodically requesting new representations (every couple of minutes) the Coyote API should be able to handle some load?

**Comment by flipsasser** [comment]
The switch to Gcloud has allowed us to add elastic scaling. We've already covered this in other issues, so I'ma close this one out and if we see something similar happen we'll track it against the new infrastructure.
